### DISCUSSION TITLE
Not able to build rabbitmq rpm from source code

### DISCUSSION THREAD

**Discussion by Vishnu-0809** [rabbitmq/rabbitmq-server#11775] [Questions] [answered]
Hi, using this git https://github.com/rabbitmq/rabbitmq-packaging and referring to rabbitmq source code as a tar.xz file from https://github.com/rabbitmq/rabbitmq-server/releases/tag/v3.13.3 I am able to build the rabbitmq rpm and use it for my purpose.

But when I follow the same procedure by cloning the rabbitmq source code from https://github.com/rabbitmq/rabbitmq-server/tree/v3.13.x make it into a tar.xz file and using rabbitmq-packaging similar to above one the build fails.

In https://github.com/rabbitmq/rabbitmq-server/releases/tag/v3.13.3 there is section called "Source Code Archives" which tells that to get the source code use this tar.xz file, so I wanted to understand that 
1. will there be any difference between in using tar.xz file in https://github.com/rabbitmq/rabbitmq-server/releases/tag/v3.13.3 and cloning the git from https://github.com/rabbitmq/rabbitmq-server/tree/v3.13.x for source code, which is failing the rabbitmq rpm build
2. If I have to refer to source code should I always use this tar.xz file from https://github.com/rabbitmq/rabbitmq-server/releases/tag/v3.13.3

Thanks in advance.

**Comment by Vishnu-0809** [comment]
Hi, any info on this please

**Reply by lukebakken** [reply]
Do NOT bump a discussion. People will assist you if they have time and interest. If you have a paid RabbitMQ support contract, use the official support channel.

**Comment by michaelklishin** [comment, marked as answer]
@Vishnu-0809 if you decide to build your own packages from source, the license allows you to do that but why do you expect the core team to be holding your hand along the way? We very obviously do not want to see custom packages being used in the wild unless you package RabbitMQ for a certain supported distribution (read: not CentOS 7 or 8), and we will not spend our time supporting such packages in any way.

You build it? You troubleshoot it. You distribute it. You maintain it.

Yes, there is a difference between a source tarball and what's in the repo.
Release notes say very specifically what should be used.

In the source repo,

``` shell
gmake package-generic-unix RABBITMQ_VERSION=x.y.z
```

will produce both source and binary build packages.
